### PR TITLE
Change from menu item to intention action

### LINKED ISSUE #9: Use an intention action instead of a menu item
I feel like an intention action (alt-enter) would make more sense. You can target it to only autovalue classes. This is also what the android studio's parcelable generator uses.

### PR DESCRIPTION
**Pull request by evant**
The intention action will only show when in an AutoValue class and it
does not have a builder or the builder needs to be updated. Fixes #9.

I have also updated it to remove methods from the builder if they are
removed from the target class.

### CODE REVIEW

File: test/AutoValuePluginTest.java [hunk 1]
```diff
@@ -16,7 +17,7 @@
 
     @Override
     protected void setUp() throws Exception {
-        VfsRootAccess.SHOULD_PERFORM_ACCESS_CHECK = false; // TODO: a workaround for v15
+//        VfsRootAccess.SHOULD_PERFORM_ACCESS_CHECK = false; // TODO: a workaround for v15
```
Comment (evant) [inline test/AutoValuePluginTest.java]: My tests wouldn't compile with this line. I wasn't sure what to do with it. Running with intellij community edition 2016.

**Comment by evant** [comment]
This is a start to make sure I'm on the right path. I would like to add intentions for factory methods as well. As well as a quick fix when the builder is not up to date.

**Comment by afcastano** [comment]
Thank you very much man!!!
I will take a quick look at it over the weekend and merge!

**Comment by afcastano** [comment]
Question:
Given that there is a fair amount of people already using the menu item, Maybe we could keep both (menu item and the intention) for a while and find a way to let them know that the menu item will be removed... It would be a bit awkward, but at least current users won't be surprised when the don't find the menu item. 

What do you think?

File: test/AutoValuePluginTest.java [hunk 2]
```diff
@@ -27,88 +28,93 @@ protected String getTestDataPath() {
...
 
     public void testAddNewProperty() {
-        runAddMissingMethodsTest("generatebuilder/addnewproperty/Test.java",
+        runIntentionAction(new UpdateBuilderAction(), "generatebuilder/addnewproperty/Test.java",
                 "generatebuilder/addnewproperty/Test_expected.java",
                 AUTOVALUE);
 
     }
 
     public void testAddBuilderFactory() {
```
Comment (afcastano) [inline test/AutoValuePluginTest.java]: This test is not passing

**Comment by afcastano** [comment]
If I use the intention when the class is not abstract it throws an array index out of bounds

**Comment by afcastano** [comment]
I will merge your pull request and do the fixes!

Also will add an action on the generate menu, so the auto value generator will work with both: Generate context menu and the intentions.

Thanks!

**Pull request merged**